### PR TITLE
HTTPExecutor now checks for status on different methods. - closes #181

### LINKED ISSUE #181: Non-HEAD HTTPExecutor Requests
The HTTPExecutor currently uses HEAD requests to execute the after_start_check. Unfortunately that is not always the right method. I have the case where HEAD requests are refused by a server that I need to start, whereas GET requests work.

I think it would therefore be a good idea to make the HTTP method configurable.

### PR DESCRIPTION
**Pull request by fizyk**
- [x] data payload

### CODE REVIEW

File: src/mirakuru/http.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ class HTTPExecutor(TCPExecutor):
     DEFAULT_PORT = 80
     """Default TCP port for the HTTP protocol."""
 
-    def __init__(self, command, url, status=r'^2\d\d$', **kwargs):
+    def __init__(self, command, url, status=r'^2\d\d$', method='HEAD', **kwargs):
```
Comment (fizyk) [inline src/mirakuru/http.py]: also might need payload

File: src/mirakuru/http.py [hunk 2]
```diff
@@ -76,7 +81,7 @@ def after_start_check(self):
         try:
             conn = HTTPConnection(self.host, self.port)
 
-            conn.request('HEAD', self.url.path)
```
Comment (fizyk) [inline src/mirakuru/http.py]: https://docs.python.org/3/library/http.client.html#examples for the POST request

File: src/mirakuru/http.py [hunk 3]
```diff
@@ -75,8 +83,19 @@ def after_start_check(self):
         """Check if defined URL returns expected status to a HEAD request."""
         try:
             conn = HTTPConnection(self.host, self.port)
-
-            conn.request('HEAD', self.url.path)
+            if self.payload and self.method in ('POST', 'PUT', 'PATCH'):
+                headers = {
+                    "Content-type": "application/x-www-form-urlencoded",
```
Comment (jotes) [inline src/mirakuru/http.py]: If users can send a custom payload then maybe they should be able to specify the `Content-type` too?

Comment (fizyk) [reply src/mirakuru/http.py]: I'm barely familiar with HTTPConnection, but it seems no headers are needed here.

But good point. Custom headers might also bee needed.

**Pull request merged**